snuba/query/parser/__init__.py

```diff
@@ -373,6 +374,38 @@ def _expand_aliases(query: Query) -> None:
 ARRAYJOIN_FUNCTION_MATCH = FunctionCallMatch(StringMatch("arrayJoin"), None)
 
 
+def _mangle_aliases(query: Query) -> None:
+    alias_prefix = "_snuba_"
+
+    def transform_alias(alias: Optional[str]) -> Optional[str]:
+        if alias is not None:
+            return f"{alias_prefix}{alias}"
+
+        return alias
+
+    query.transform_expressions(
+        lambda expr: replace(expr, alias=transform_alias(expr.alias))
```

Comment:
You do not need to rebuild expressions where the alias is not present. You can simply transform the alias, and execute the replace only if it changes (that saves allocation since this happens on the full query).